lib/tts_isolate.dart

```diff
@@ -59,28 +61,44 @@ void onStart(ServiceInstance service) async {
 
   service.on('initSharedPreference').listen((event) async {
     final prefs = await StreamingSharedPreferences.instance;
-    prefs.getString('tts_channel', defaultValue: '{}').listen((channel) async {
-      debugPrint("tts_channel changed to: $channel");
-      if (channel.isNotEmpty && channel != "{}") {
-        // Fetch messages from Firestore
-        // var messages = await fetchMessagesFromFirestore(channel);
-        // print('MESSAGES HERE $messages');
-        // Process messages as needed
-      }
-    });
+
+    prefs
+        .getString('tts_channel', defaultValue: '{}')
+        .switchMap((channel) {
+          if (channel.isNotEmpty && channel != "{}") {
+            return FirebaseFirestore.instance
+                .collection('channels')
+                .where('channelId', isEqualTo: channel)
+                .snapshots();
+          } else {
+            return Stream.empty();
+          }
+        })
+        .switchMap((snapshot) => Stream.fromIterable(snapshot.docChanges))
```

Comment:
Ah ok, so I was a little confused. It's not this change, but rather `service.on('initSharedPreferences')` isn't correct.

Actually, `flutter_background_service` already does the isolate handling for you, so all that needs to be done is instead of calling `service.on('...'), at the top level just add the listener. That is,

```
await Firebase.initializeApp();

final prefs = await ...;

prefs.getString(...).switchMap(...)...
```

I'd strongly recommend actually testing this by clicking the button and getting the tts engine to spit out text. This stuff is not easy to figure out without trying it.

Reply:
Just pushed some changes, @SputNikPlop is going to test!